[cmake] FindFlatbuffers enable detecting existing cmake config -> not working with `FLATBUFFERS_BUILD_FLATLIB=ON`

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Since PR https://github.com/xbmc/xbmc/pull/24981 in cmake file compile with FlatBuffers `FLATBUFFERS_BUILD_FLATLIB=ON` is broken.
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->



## Expected Behavior
It should compile with `FLATBUFFERS_BUILD_FLATLIB=OFF` AND `FLATBUFFERS_BUILD_FLATLIB=ON` like before.
<!--- Tell us what should happen -->
<!--- Put your text below this line -->

### To Reproduce
Steps to reproduce the behavior:
Enable `FLATBUFFERS_BUILD_FLATLIB=ON` in flatbuffer external package:
https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/devel/flatbuffers/package.mk#L26
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->

### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
```
-- Found FlatBuffers: /home/portisch/CE/CoreELEC/build.CoreELEC-Amlogic-ne.aarch64-22/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/include/ (found version "23.3.3") 
CMake Error at cmake/modules/FindFlatBuffers.cmake:59 (add_library):
  add_library cannot create imported target "flatbuffers::flatbuffers"
  because another target with the same name already exists.
```
### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 CoreELEC-22, same as LibreELEC/master
 - Kodi version:
xbmc/master


<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/cmake/modules/FindFlatBuffers.cmake
+++ b/cmake/modules/FindFlatBuffers.cmake
@@ -4,11 +4,11 @@
 #
 # This will define the following target:
 #
-#   flatbuffers::flatbuffers - The flatbuffers headers
+#   flatbuffers::flatheaders - The flatbuffers headers
 
 find_package(FlatC REQUIRED)
 
-if(NOT TARGET flatbuffers::flatbuffers)
+if(NOT TARGET flatbuffers::flatheaders)
 
   include(cmake/scripts/common/ModuleHelpers.cmake)
 
@@ -56,14 +56,14 @@ if(NOT TARGET flatbuffers::flatbuffers)
                                     REQUIRED_VARS FLATBUFFERS_INCLUDE_DIR
                                     VERSION_VAR FLATBUFFERS_VER)
 
-  add_library(flatbuffers::flatbuffers INTERFACE IMPORTED)
-  set_target_properties(flatbuffers::flatbuffers PROPERTIES
+  add_library(flatbuffers::flatheaders INTERFACE IMPORTED)
+  set_target_properties(flatbuffers::flatheaders PROPERTIES
                                                  INTERFACE_INCLUDE_DIRECTORIES "${FLATBUFFERS_INCLUDE_DIR}")
 
-  add_dependencies(flatbuffers::flatbuffers flatbuffers::flatc)
+  add_dependencies(flatbuffers::flatheaders flatbuffers::flatc)
 
   if(TARGET flatbuffers)
-    add_dependencies(flatbuffers::flatbuffers flatbuffers)
+    add_dependencies(flatbuffers::flatheaders flatbuffers)
   endif()
 
   # Add internal build target when a Multi Config Generator is used
@@ -82,5 +82,5 @@ if(NOT TARGET flatbuffers::flatbuffers)
     add_dependencies(build_internal_depends flatbuffers)
   endif()
 
-  set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP flatbuffers::flatbuffers)
+  set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP flatbuffers::flatheaders)
 endif()

--- a/cmake/modules/FindFlatBuffers.cmake
+++ b/cmake/modules/FindFlatBuffers.cmake
@@ -56,31 +56,38 @@ if(NOT TARGET flatbuffers::flatheaders)
                                     REQUIRED_VARS FLATBUFFERS_INCLUDE_DIR
                                     VERSION_VAR FLATBUFFERS_VER)
 
-  add_library(flatbuffers::flatheaders INTERFACE IMPORTED)
-  set_target_properties(flatbuffers::flatheaders PROPERTIES
-                                                 INTERFACE_INCLUDE_DIRECTORIES "${FLATBUFFERS_INCLUDE_DIR}")
+  if(FlatBuffers_FOUND)
 
-  add_dependencies(flatbuffers::flatheaders flatbuffers::flatc)
+    add_library(flatbuffers::flatheaders INTERFACE IMPORTED)
+    set_target_properties(flatbuffers::flatheaders PROPERTIES
+                                                   INTERFACE_INCLUDE_DIRECTORIES "${FLATBUFFERS_INCLUDE_DIR}")
 
-  if(TARGET flatbuffers)
-    add_dependencies(flatbuffers::flatheaders flatbuffers)
-  endif()
+    add_dependencies(flatbuffers::flatheaders flatbuffers::flatc)
 
-  # Add internal build target when a Multi Config Generator is used
-  # We cant add a dependency based off a generator expression for targeted build types,
-  # https://gitlab.kitware.com/cmake/cmake/-/issues/19467
-  # therefore if the find heuristics only find the library, we add the internal build
-  # target to the project to allow user to manually trigger for any build type they need
-  # in case only a specific build type is actually available (eg Release found, Debug Required)
-  # This is mainly targeted for windows who required different runtime libs for different
-  # types, and they arent compatible
-  if(_multiconfig_generator)
-    if(NOT TARGET flatbuffers)
-      buildflatbuffers()
-      set_target_properties(flatbuffers PROPERTIES EXCLUDE_FROM_ALL TRUE)
+    if(TARGET flatbuffers)
+      add_dependencies(flatbuffers::flatheaders flatbuffers)
     endif()
-    add_dependencies(build_internal_depends flatbuffers)
-  endif()
 
-  set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP flatbuffers::flatheaders)
+    # Add internal build target when a Multi Config Generator is used
+    # We cant add a dependency based off a generator expression for targeted build types,
+    # https://gitlab.kitware.com/cmake/cmake/-/issues/19467
+    # therefore if the find heuristics only find the library, we add the internal build
+    # target to the project to allow user to manually trigger for any build type they need
+    # in case only a specific build type is actually available (eg Release found, Debug Required)
+    # This is mainly targeted for windows who required different runtime libs for different
+    # types, and they arent compatible
+    if(_multiconfig_generator)
+      if(NOT TARGET flatbuffers)
+        buildflatbuffers()
+        set_target_properties(flatbuffers PROPERTIES EXCLUDE_FROM_ALL TRUE)
+      endif()
+      add_dependencies(build_internal_depends flatbuffers)
+    endif()
+
+    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP flatbuffers::flatheaders)
+  else()
+    if(FlatBuffers_FIND_REQUIRED)
+      message(FATAL_ERROR "Flatbuffer schema headers were not found. You may want to try -DENABLE_INTERNAL_FLATBUFFERS=ON to build the internal headers package")
+    endif()
+  endif()
 endif()

--- a/xbmc/cores/RetroPlayer/messages/CMakeLists.txt
+++ b/xbmc/cores/RetroPlayer/messages/CMakeLists.txt
@@ -20,8 +20,8 @@ set_target_properties(retroplayer_messages PROPERTIES FOLDER "Generated Messages
                                                       INCLUDE_DIRECTORIES ${CMAKE_CURRENT_BINARY_DIR}
                                                       SOURCES "${FLATC_OUTPUTS}")
 
-if(TARGET flatbuffers::flatbuffers)
-  set_property(TARGET flatbuffers::flatbuffers APPEND PROPERTY
+if(TARGET flatbuffers::flatheaders)
+  set_property(TARGET flatbuffers::flatheaders APPEND PROPERTY
                                                INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_BINARY_DIR}")
-  add_dependencies(retroplayer_messages flatbuffers::flatbuffers)
+  add_dependencies(retroplayer_messages flatbuffers::flatheaders)
 endif()

